### PR TITLE
Standardized exception messages

### PR DESCRIPTION
**Pull request by allanbz**
All exceptions messages now have the first character capitalized and never end in punctuation. Hope I was helpfull.

### CODE REVIEW

File: src/main/java/org/mafagafogigante/dungeon/util/Table.java [hunk 1]
```diff
@@ -157,9 +157,9 @@ public void insertRow(String... values) {
     if (values.length != columns.size()) {
       String expectedButGotString = "Expected " + columns.size() + ", but got " + values.length + ".";
```
Comment (mafagafogigante) [inline src/main/java/org/mafagafogigante/dungeon/util/Table.java]: This ends up adding a dot at the end of the message.

File: src/main/java/org/mafagafogigante/dungeon/schema/rules/BoundIntegerJsonRule.java [hunk 2]
```diff
@@ -17,10 +17,10 @@ public void validate(JsonValue value) {
     super.validate(value);
     int intValue = value.asInt();
     if (intValue < minValue) {
-      throw new IllegalArgumentException(value + " is below the allowed minimum " + minValue + ".");
+      throw new IllegalArgumentException(value + " is below the allowed minimum " + minValue);
```
Comment (mafagafogigante) [inline src/main/java/org/mafagafogigante/dungeon/schema/rules/BoundIntegerJsonRule.java]: It should actually be "the allowed minimum OF ...", my mistake. Let's fix this now.

File: src/main/java/org/mafagafogigante/dungeon/schema/rules/BoundIntegerJsonRule.java [hunk 3]
```diff
@@ -17,10 +17,10 @@ public void validate(JsonValue value) {
...
     }
     if (intValue > maxValue) {
-      throw new IllegalArgumentException(value + " is above the allowed maximum " + maxValue + ".");
+      throw new IllegalArgumentException(value + " is above the allowed maximum " + maxValue);
```
Comment (mafagafogigante) [inline src/main/java/org/mafagafogigante/dungeon/schema/rules/BoundIntegerJsonRule.java]: It should actually be "the allowed maximum OF ...", my mistake.

File: src/main/java/org/mafagafogigante/dungeon/schema/rules/StringLengthJsonRule.java [hunk 4]
```diff
@@ -15,10 +15,10 @@ public void validate(JsonValue value) {
     super.validate(value);
     String stringValue = value.asString();
     if (stringValue.length() < stringLength) {
-      throw new IllegalArgumentException(value + " is below the allowed minimum " + stringLength + ".");
+      throw new IllegalArgumentException(value + " is below the allowed minimum " + stringLength);
```
Comment (mafagafogigante) [inline src/main/java/org/mafagafogigante/dungeon/schema/rules/StringLengthJsonRule.java]: It should actually be "the allowed minimum OF ...", my mistake.

File: src/main/java/org/mafagafogigante/dungeon/schema/rules/StringLengthJsonRule.java [hunk 5]
```diff
@@ -15,10 +15,10 @@ public void validate(JsonValue value) {
...
     }
     if (stringValue.length() > stringLength) {
-      throw new IllegalArgumentException(value + " is above the allowed maximum " + stringLength + ".");
+      throw new IllegalArgumentException(value + " is above the allowed maximum " + stringLength);
```
Comment (mafagafogigante) [inline src/main/java/org/mafagafogigante/dungeon/schema/rules/StringLengthJsonRule.java]: It should actually be "the allowed maximum OF ...", my mistake.

File: src/main/java/org/mafagafogigante/dungeon/achievements/AchievementTracker.java [hunk 6]
```diff
@@ -78,7 +78,7 @@ private UnlockedAchievement getUnlockedAchievement(Achievement achievement) {
    */
   public List<UnlockedAchievement> getUnlockedAchievements(Comparator<UnlockedAchievement> comparator) {
     if (comparator == null) {
-      throw new IllegalArgumentException("comparator is null.");
+      throw new IllegalArgumentException("Comparator is null");
```
Comment (mafagafogigante) [inline src/main/java/org/mafagafogigante/dungeon/achievements/AchievementTracker.java]: We should get rid of the check and use a `@NotNull` annotation instead.

File: src/main/java/org/mafagafogigante/dungeon/commands/CommandDescription.java [hunk 7]
```diff
@@ -20,7 +20,7 @@
    */
   public CommandDescription(String name, String info) {
     if (name == null) {
-      throw new IllegalArgumentException("Cannot create CommandDescription with null name.");
+      throw new IllegalArgumentException("Cannot create CommandDescription with null name");
```
Comment (mafagafogigante) [inline src/main/java/org/mafagafogigante/dungeon/commands/CommandDescription.java]: We should get rid of the check and use a `@NotNull` annotation instead.

Back when I wrote this I was not using the annotation.

**Comment by mafagafogigante** [comment]
@allanbz Finished what I pointed out and got this into the branch seven today, closing because I rebase instead of merging to keep the history clear. I've also added you to the contributors list. Thanks.

**Comment by allanbz** [comment]
Sorry, I was planning to fix what you pointed out, but I got busy with college projects and it turned out I did not have time to help. Thanks for adding me to the contributors, it means a lot to me. I started using Java a short time ago, and your code helped me learn a few things.

**Comment by mafagafogigante** [comment]
@allanbz No problems, if you ever feel like practicing more in a relatively large (10K SLoC) and high quality project, feel free to submit a PR.

**Pull request closed without merging**